src/lib/Microsoft.Health.Fhir.Ingest.Template/CollectionContentTemplateFactory.cs

```diff
@@ -44,11 +45,27 @@ protected override IContentTemplate BuildCollectionTemplate(JArray templateColle
                 }
                 catch (InvalidTemplateException ex)
                 {
-                    errors.Add(new TemplateError(ex.Message));
+                    errors.Add(new TemplateError(ex.Message, ex.GetLineInfo));
                 }
                 catch (JsonSerializationException ex)
+                {
+                    errors.Add(new TemplateError(ex.Message, new LineInfo() { LineNumber = ex.LineNumber, LinePosition = ex.LinePosition }));
+                }
+                catch (AggregateException ex)
                 {
                     errors.Add(new TemplateError(ex.Message));
+
+                    foreach (var innerException in ex.InnerExceptions)
```

Comment:
Suggestion: Code is repeated a couple times.  Would be good to extract into an extension or a common path.

Reply:
I'll throw this in an extension method.